File: gild-model/InventoryItem.cs
using Newtonsoft.Json;
using System;

namespace gild_model
{
    public class InventoryItem
    {
        [JsonProperty("id")]
        public Guid Id { get; set; } = Guid.NewGuid();

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("category")]
        public string Category { get; set; }

        [JsonProperty("sellIn")]
        public decimal SellIn { get; set; }

        [JsonProperty("quality")]
        public decimal Quality { get; set; }        
    }
}

File: apps/grose-ui/src/components/InventoryItem.js
import React, { useState, useEffect } from 'react';

const InventoryItem = props => {
    const [item, updateItem] = useState(props);

    useEffect(() => {
        updateItem(props);
    }, [props]);

    return (
        <div className="InventoryItem">
            <div>{item.name}</div>
            <div>{item.category}</div>
            <div>{item.sellIn}</div>
            <div>{item.quality}</div>
        </div>
    );
}

export default InventoryItem;